### PR TITLE
[ALLUXIO-2252] Fix master fault tolerance integration test

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2252

### CODE REVIEW

File: core/server/src/main/java/alluxio/master/AbstractMaster.java [hunk 1]
```diff
@@ -64,14 +67,14 @@
   /**
    * @param journal the journal to use for tracking master operations
    * @param clock the Clock to use for determining the time
-   * @param executorService the executor service to use for running maintenance threads; the
-   *        {@link AbstractMaster} becomes the owner of the executorService and will shut it down
-   *        when the master stops
+   * @param executorServiceFactory a factory for creating the the executor service to use for
```
Comment (gpang) [inline core/server/src/main/java/alluxio/master/AbstractMaster.java]: NIT: repeated "the"

File: core/server/src/main/java/alluxio/master/AbstractMaster.java [hunk 2]
```diff
@@ -174,16 +179,19 @@ public void stop() throws IOException {
...
+        if (!mExecutorService.awaitTermination(SHUTDOWN_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+          LOG.warn("Timed out " + awaitFailureMessage, this.getClass().getSimpleName());
+        }
+      } catch (InterruptedException e) {
+        LOG.warn("Interrupted while " + awaitFailureMessage, this.getClass().getSimpleName());
       }
-    } catch (InterruptedException e) {
-      LOG.warn("Interrupted while " + awaitFailureMessage, this.getClass().getSimpleName());
     }
+    mExecutorService = null;
```
Comment (gpang) [inline core/server/src/main/java/alluxio/master/AbstractMaster.java]: Can anything above (lines 183 - 191) throw any other type of exception which might skip over this line that sets it to `null`?

File: tests/src/test/java/alluxio/master/MasterFaultToleranceIntegrationTest.java [hunk 3]
```diff
@@ -58,9 +56,9 @@ public final void before() throws Exception {
     // TODO(gpang): Implement multi-master cluster as a resource.
     mMultiMasterLocalAlluxioCluster =
         new MultiMasterLocalAlluxioCluster(MASTERS);
+    mMultiMasterLocalAlluxioCluster.initConfiguration();
```
Comment (gpang) [inline tests/src/test/java/alluxio/master/MasterFaultToleranceIntegrationTest.java]: why was this moved?

Comment (aaudiber) [inline tests/src/test/java/alluxio/master/MasterFaultToleranceIntegrationTest.java]: `initConfiguration()` sets most Alluxio configuration to default values, so it needs to happen before `Configuration.set`, or it will override the values we want.

File: core/server/src/main/java/alluxio/master/AbstractMaster.java (hunk 1 above)
Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/AbstractMaster.java]: fixed, thanks

File: core/server/src/main/java/alluxio/master/AlluxioMaster.java [hunk 4]
```diff
@@ -442,6 +435,10 @@ protected void startServing(String startMessage, String stopMessage) {
   }
 
   protected void startServingWebServer() {
+    mWebServer = new MasterUIWebServer(ServiceType.MASTER_WEB.getServiceName(),
```
Comment (gpang) [inline core/server/src/main/java/alluxio/master/AlluxioMaster.java]: why was this block moved?

Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/AlluxioMaster.java]: the `stop` method shuts the web server down and sets `mWebServer` to null, so the `start` method needs to create the web server

File: core/common/src/main/java/alluxio/util/executor/ExecutorServiceFactories.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+ * Convenience methods for constructing instances of {@link ExecutorServiceFactory}.
+ */
+public final class ExecutorServiceFactories {
+  /**
+   * @param name the base name for executor thread names
+   * @param nThreads the number of threads to create executors with
+   * @return an {@link ExecutorServiceFactory} which creates threadpool executors with the given
+   *         name and number of threads
+   */
+  public static ExecutorServiceFactory fixedThreadPoolExecutorServiceFactory(final String name,
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/util/executor/ExecutorServiceFactories.java]: Should indicate that this is executor service will use daemon threads?

File: core/server/src/main/java/alluxio/master/AlluxioMaster.java (hunk 4 above)
Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/AlluxioMaster.java]: The web server life cycle is no longer in sync with the other servers?

File: core/common/src/main/java/alluxio/util/executor/ExecutorServiceFactories.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+import alluxio.util.ThreadFactoryUtils;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Convenience methods for constructing instances of {@link ExecutorServiceFactory}.
+ */
+public final class ExecutorServiceFactories {
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/util/executor/ExecutorServiceFactories.java]: This eventually will be used everywhere?

File: core/server/src/main/java/alluxio/master/AlluxioMaster.java (hunk 4 above)
Comment (aaudiber) [inline core/server/src/main/java/alluxio/master/AlluxioMaster.java]: Right. I'm going for the path of least resistance here on getting master fault tolerance fixed. I think we definitely need a followup PR which makes sense of all the different lifecycle.

File: core/common/src/main/java/alluxio/util/executor/ExecutorServiceFactories.java (hunk 6 above)
Comment (aaudiber) [inline core/common/src/main/java/alluxio/util/executor/ExecutorServiceFactories.java]: It's convenient for avoiding messy syntax anywhere that we need an `ExecutorServiceFactory`.

File: core/server/src/main/java/alluxio/master/AlluxioMaster.java (hunk 4 above)
Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/AlluxioMaster.java]: Got it, thanks.

File: core/common/src/main/java/alluxio/util/executor/ExecutorServiceFactories.java (hunk 5 above)
Comment (aaudiber) [inline core/common/src/main/java/alluxio/util/executor/ExecutorServiceFactories.java]: Updated

**Comment by calvinjia** [comment]
LGTM
